packages/experience/src/pages/Register/index.tsx

```diff
@@ -65,6 +58,37 @@ const Register = () => {
           </>
         )
       }
+    </>
+  );
+};
+
+const Register = () => {
+  const { signUpMethods, socialConnectors, signInMode } = useSieMethods();
+
+  if (!signInMode) {
+    return <ErrorPage />;
+  }
+
+  if (signInMode === SignInMode.SignIn) {
+    return <Navigate to="/sign-in" />;
+  }
+
+  return (
+    <LandingPageLayout title="description.create_your_account">
+      <SingleSignOnFormModeContextProvider>
+        {signUpMethods.length > 0 && (
+          <IdentifierRegisterForm signUpMethods={signUpMethods} className={styles.main} />
+        )}
+        {signUpMethods.length === 0 && socialConnectors.length > 0 && (
+          <>
+            <TermsAndPrivacy className={styles.terms} />
+            <SocialSignInList className={styles.main} socialConnectors={socialConnectors} />
+          </>
+        )}
+        <RegisterFooter />
```

Comment:
extract all the content into a sub component. If is single sign on form mode, return null.